signac/core/synced_collections/utils.py

```diff
@@ -141,3 +141,64 @@ def default(self, o: Any) -> Dict[str, Any]:  # noqa: D102
             # Call the super method, which raises a TypeError if it cannot
             # encode the object.
             return super().default(o)
+
+
+class _NullContext:
+    """A nullary context manager.
+
+    There are various cases where we sometimes want to perform a task within a
+    particular context, but at other times we wish to ignore that context. The
+    most obvious example is a lock for threading: since
+    :class:`SyncedCollection`s allow multithreading support to be enabled or
```

Comment:
To pluralize a class reference in rST, you either need a raw string with an escaped space like:
```python
r"""This is how to pluralize :class:`SyncedCollection`\ s."""
```
or (my preference) you can just treat the class as a collective/singular noun.
```suggestion
    :class:`SyncedCollection` allows multithreading support to be enabled or
```